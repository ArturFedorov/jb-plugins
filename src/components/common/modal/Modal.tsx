import React, { FunctionComponent, ReactElement } from 'react';
import classNames from 'classnames';
import styles from './modal.module.scss';

export interface IModalProps {
  showModal?: boolean;
  header: string;
  message: string;
  onClose: () => void;
  children?: ReactElement<Element>[];
}

export const Modal: FunctionComponent<IModalProps> = ({
  showModal,
  header,
  message,
  onClose,
  children
}) => {
  return (
    <div>
      {showModal && (
        <div className={styles.module} onClick={onClose}>
          <div className={styles.moduleContent}>
            <h3 className={classNames(styles.moduleHeader, 'is-lighter')}>{header}</h3>
            <p className="is-gray-text">{message}</p>
            <div className={classNames(styles.loadingBar, styles.loadingBarLoading)}></div>
            <div className={styles.moduleButtons}>
              {children}
            </div>
          </div>
        </div>
      )}
    </div>
  );
};
