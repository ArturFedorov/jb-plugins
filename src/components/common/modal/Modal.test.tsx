import React from 'react';
import { shallow } from 'enzyme';
import { IModalProps, Modal } from './Modal';

describe('Modal', () => {
  const defaultProps: IModalProps = {
    header: 'Test header',
    message: 'Test message',
    onClose: jest.fn()
  };

  it('should match a snapshot', () => {
    const modal = shallow(<Modal {...defaultProps} />);
    expect(modal).toMatchSnapshot();
  });
});
