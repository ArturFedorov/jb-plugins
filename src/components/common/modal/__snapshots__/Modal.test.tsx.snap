// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match a snapshot 1`] = `ShallowWrapper {}`;
